slice/internal/controller/workload_controller.go

```diff
@@ -361,46 +420,78 @@ func (r *WorkloadReconciler) sliceAC(ctx context.Context, wl *kueue.Workload) (*
 	return workload.FindAdmissionCheck(wl.Status.AdmissionChecks, relevantChecks[0]), nil
 }
 
-func parseTopologyAssignmentIntoNodeSelector(slice *v1alpha1.Slice, wl *kueue.Workload) {
-	nodeSelectors := sets.New[string]()
+func (r *WorkloadReconciler) syncSlices(ctx context.Context, wl *kueue.Workload, ac *kueue.AdmissionCheckState, slices *[]v1alpha1.Slice) error {
+	slicesByName := make(map[string]*v1alpha1.Slice, len(*slices))
+	for _, slice := range *slices {
+		slicesByName[slice.Name] = &slice
+	}
+
+	capacity := len(wl.Status.Admission.PodSetAssignments) - len(slicesByName)
+	if capacity < 0 {
+		capacity = 0
+	}
+	createdSlices := make([]v1alpha1.Slice, 0, capacity)
+
 	for _, psa := range wl.Status.Admission.PodSetAssignments {
-		// we already validated that all assignments have a valid level,
-		// in validateRelevantWorkload.
-		subblockLevelIndex := topology.SubblockLevelIndex(&psa)
-		for _, domain := range psa.TopologyAssignment.Domains {
-			nodeSelectors.Insert(domain.Values[subblockLevelIndex])
+		sliceName := core.SliceName(wl.Name, psa.Name)
```

Comment:
In follow-up PR:

Should we use `IsRelevantPodTemplateSpec` in this loop? Otherwise, we may try to create a slice for an invalid podset. Can you add a test for this to please?